src/classes/PReq_CannotDelete_TDTM.cls

```diff
@@ -35,30 +35,39 @@
 * @description Stops an Plan Requirement from being deleted if it has any child Plan Requirement
 */
 public with sharing class PReq_CannotDelete_TDTM extends TDTM_Runnable {
-    /*******************************************************************************************************
-    * @description Stops an Plan Requirement from being deleted if it has any child Plan Requirement
-    * @param listNew the list of Plan Requirement from trigger new.
-    * @param listOld the list of Plan Requirement from trigger old.
-    * @param triggerAction which trigger event (BeforeInsert, AfterInsert, etc.).
-    * @param objResult the describe for Plan Requirement
-    * @return dmlWrapper.
-    ********************************************************************************************************/
-    public override DmlWrapper run(List<SObject> newlist, List<SObject> oldlist,
-        TDTM_Runnable.Action triggerAction, Schema.DescribeSObjectResult objResult) {
+  /*******************************************************************************************************
+  * @description Get the setting of preventing Account deletion
+  */
+  private static Boolean enabledPreventPlanRequirementDeletion = UTIL_CustomSettingsFacade.getSettings().Prevent_Plan_Requirement_Deletion__c;
+  
+  /*******************************************************************************************************
+  * @description Stops an Plan Requirement from being deleted if it has any child Plan Requirement
+  * @param listNew the list of Plan Requirement from trigger new.
+  * @param listOld the list of Plan Requirement from trigger old.
+  * @param triggerAction which trigger event (BeforeInsert, AfterInsert, etc.).
+  * @param objResult the describe for Plan Requirement
+  * @return dmlWrapper.
+  ********************************************************************************************************/
+  public override DmlWrapper run(List<SObject> newlist, List<SObject> oldlist, 
```

Comment:
These present an opportunity for us to split this out into separate methods to assist with refactoring and focused unit tests. In the future, I'd like us to consider minimizing the amount of logic we have in run methods.